uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -12,6 +16,28 @@ function tasksassignments_civicrm_config(&$config) {
 }
 
 /**
+ * @param array $wrappers
+ * @param array $apiRequest
+ */
+function tasksassignments_civicrm_apiWrappers(&$wrappers, $apiRequest) {
+  $requestJson = ArrayHelper::value('json', $_REQUEST);
+  $params = ArrayHelper::value('params', $apiRequest, []);
+  $optionGroup = ArrayHelper::value('option_group_id', $params);
+  $isActivityType = $optionGroup === 'activity_type';
+  $entity = ArrayHelper::value('entity', $apiRequest);
+  $action = ArrayHelper::value('action', $apiRequest);
+
+  $jsonFromCaseEdit = '"actTypes":["OptionValue","get",{"option_group_id":"activity_type","options":{"sort":"name","limit":0}}]';
+  $isAssignmentEdit = strpos($requestJson, $jsonFromCaseEdit) !== FALSE;
+
+  if ($isAssignmentEdit && $entity === 'OptionValue' && $action === 'get' && $isActivityType) {
+    $wrappers[] = new StripInvalidTaskTypesWrapper();
+  } elseif ($entity === 'CaseType' && $action === 'create') {
```

Comment:
> I would say that just hiding it is not enough. If we don't want to allow it then we should have validation to stop it.

I asked for that because the way in which you prevent creating it is a but dangerous , it apply to 'CaseType' create actions .. but hrrequirement extensions has some usages for CaseType entity  so I don't want to risk that until the relationship between them is more clear.

> Also for some people who have already created assignments we should tell them if they already have activity types that are no valid.

I don't think this a big problem now .. applying what you did to 'create' CaseType entity worrying me more .